mavros/include/mavros/frame_tf.h

```diff
@@ -309,5 +336,28 @@ inline Eigen::Quaterniond mavlink_to_quaternion(const std::array<float, 4> &q)
 	return Eigen::Quaterniond(q[0], q[1], q[2], q[3]);
 }
 
+/**
+ * @brief Store Covariance matrix to MAVLink float[n] format
+ */
+template<class T, std::size_t SIZE>
+inline void covariance_to_mavlink(const T &cov, std::array<float, SIZE> &covmsg) {
+	std::copy(cov.cbegin(), cov.cend(), covmsg.begin());
+}
+
+/**
+ * @brief Store half upper right triangular of 9d Covariance matrix to MAVLink float[n] format
+ */
+template<class T, std::size_t SIZE>
+inline void covariance9d_urt_to_mavlink(const T &cov, std::array<float, SIZE> &covmsg) {
+	std::vector<float> aux;
+	size_t n = 9;
+	for (size_t i=0; i < n; i++) {
+		for (size_t j=i; j < n; j++) {
+			aux.push_back(cov.data()[i + j * n]);
+		}
+	}
+	std::copy(aux.cbegin(), aux.cend(), covmsg.begin());
```

Comment:
It was the most straighforward way found. Do you have another suggestion, so this becomes generic and not only for 9d matrices?